### ISSUE TITLE
Add missing "age related" fields to the Search All feature

### ISSUE THREAD

**Issue by jberube** [kids-first/kf-portal-ui#1459]
Follow-up of issue #1368 

The remaining missing fields are the numeric fields (age), as we cannot do aggregations on them.

    // Clinical
    // 'diagnoses.age_at_event_days',
    // 'outcome.age_at_event_days',
    // 'phenotype.age_at_event_days',

    // Biospecimen
    // 'biospecimens.age_at_event_days',
Adding one of those fields would return that error:

"Cannot query field "buckets" on type "NumericAggregations"."

~~An alternative way of fetching results should be used on them, using interval, e.g. : src/components/CohortBuilder/Summary/AgeDiagChart.js.~~

The numeric fields can be excluded from the query, and the UI can be similar (or re-use) the RangeFilter.

To build the resulting sqon, we will also need to take the type of field in consideration. It may be possible to extract it from the RangeFilter.

**Comment by jberube** [comment]
@hlminh2000 and I had a nice conversation over slack that covers the solution well.

TL:DR;

- get the `type` of each field from the `extendedMapping` already available 
- remove numeric fields from the query
- when redering a field in the filter, render the RangeFilter instead (or something akin to it).
- change the sqon generated from the numeric fields's filters

--- 

Julien Bérubé [3:01 PM]
Hi, I have to estimate this bug/limitation of the Search All: https://github.com/kids-first/kf-portal-ui/issues/1459
jberube
#1459 Add missing "age related" fields to the Search All feature
Follow-up of issue #1368

The remaining missing fields are the numeric fields (age), as we cannot do aggregations on them.

```
// Clinical
// 'diagnoses.age_at_event_days',
// 'outcome.age_at_event_days',
// 'phenotype.age_at_event_days',
```

 Show more
kids-first/kf-portal-uiMar 27thAdded by GitHub
Do you have an idea of a potential solution? I suggested something in the details of the ticket, but I honestly am not sure about it.

Minh Ha [3:03 PM]
ah, so you’ll need to pull the `extendedMapping` from arranger to identify the field type, and switch the rendered component accordingly. there’s a `ExtendedMappingProvider` you can use, example from `src/components/CohortBuilder/Categories/Category.js`

Julien Bérubé [3:04 PM]
I already use it for the count of elements

Minh Ha [3:06 PM]
you can do `<ExtendedMappingProvider field={...}/>` which will filter the result to just the field you’re interested in, memoizing that so it’s not searching everytime too
basically there, you’ll get a `type` field
that’ll tell you if it’s numeric or not

Minh Ha [3:08 PM]
oh I guess in this case the `field` prop isn’t very relevant eh

Minh Ha [3:10 PM]
but yeah basically if the gql query is constructed with a switch between numeric and keyword fields based on the `type`, that should do it

Julien Bérubé [3:20 PM]
I  won't be able to to that, right?
```
const getBuckets = searchFields =>
  searchFields
    .map(toGqlFieldPath)
    .reduce((acc, gqlFieldName) => `${acc} ${gqlFieldName} { buckets { doc_count key } }`, '');

export const searchAllFieldsQuery = (sqon, searchFields) => ({
  query: gql`
    query($sqon: JSON) {
      participant {
        aggregations(filters: $sqon) {
          ${getBuckets(searchFields)}
        }
      }
    }
  `,
```
I'm I right to presume that the all numeric fields will needs to be excluded from `aggregations` and will need to be put elsewhere?

Minh Ha [3:22 PM]
`getBuckets` can be rewritten to `getAggregation` and handle the switching there I think

Julien Bérubé [3:23 PM]
This is where my understanding of graphql falls short :stuck_out_tongue:

Minh Ha [3:23 PM]
downstream will have to switch UI too though

Julien Bérubé [3:23 PM]
Ok, so I would filter out numerics
How, right. I have to deal with range of values instead of aggregated buckets of values.
And that implies configurations for the ranges and different UI for the results.

Minh Ha [3:25 PM]
yes, that’s the idea :slightly_smiling_face:
we can reuse the range filter component I believe

Julien Bérubé [3:28 PM]
That'd be good :coffee:
What I'm thinking of is to generate a query like the one in src/components/CohortBuilder/Summary/AgeDiagChart.js:29 to :139 for each numeric field, but if you can suggest a better way, I'd be all ears!

Minh Ha [3:33 PM]
come to think of it more, I’m not sure if we really need to query anything for these fields, because the sqon should contain all the information needed to render it eh?
for the range, we don’t have to display doc counts or anything

Julien Bérubé [3:35 PM]
I still need to get some informations like the number of participants per value range.

Minh Ha [3:36 PM]
oh I see what you mean, bucketing the range and showing check boxes eh?… I’d confirm with the BAs on what’s intended there :confused:
because if we just put the range filter, that shouldn’t require any actual searching

Julien Bérubé [3:37 PM]
I think I presumed of a solution that is not the right on UX-wise
I was thinking into putting a list of ranges of values, along wth the number of participants in each. That would be homogeonous with the other results, but I get what you're talking about and it's WAY simpler!
My bad, I presumed of a solution and got my mind stuck to it :stuck_out_tongue: (edited) 
Just displaying min, max, unit, total participants is easier, better.
